code/modules/research/xenobiology/xenobio_camera.dm

```diff
@@ -119,7 +119,7 @@
 		hotkey_help.Grant(user)
 		actions += hotkey_help
 
-	RegisterSignal(user, COMSIG_MOB_CTRL_CLICKED, .proc/on_ctrl_click)
+	RegisterSignal(user, list(COMSIG_XENO_SLIME_CLICK_CTRL, COMSIG_XENO_TURF_CLICK_CTRL, COMSIG_XENO_MONKEY_CLICK_CTRL, COMSIG_MOB_CTRL_CLICKED), .proc/on_ctrl_click)
```

Comment:
You can actually probably split this up so that each signal calls their corresponding proc
This is how it is now but it'd get rid of the typechecks completely. It also means you can get rid of the COMSIG_MOB_CTRL_CLICKED as well
```dm
/obj/machinery/computer/camera_advanced/xenobio/proc/on_ctrl_click(datum/source, atom/clicked_atom)
	SIGNAL_HANDLER
	if(ismonkey(clicked_atom))
		XenoMonkeyClickCtrl(source, clicked_atom)
	if(isopenturf(clicked_atom))
		XenoTurfClickCtrl(source, clicked_atom)
	if(isslime(clicked_atom))
		XenoSlimeClickCtrl(source, clicked_atom)

Reply:
ill split it when i wake up tomorrow